ts/features/wallet/onboarding/paypal/screen/PayPalOnboardingCheckoutScreen.tsx

```diff
@@ -0,0 +1,147 @@
+  }
+) => {
+  const [isOnboardingCompleted, setIsOnboardingComplete] = useState(false);
+  const urlPrefix = props.isPagoPATestEnabled
+    ? pagoPaApiUrlPrefixTest
+    : pagoPaApiUrlPrefix;
+  return fold(
+    props.pmToken,
+    () => <LoadingOrError hasError={false} onRetry={props.refreshPMtoken} />,
+    () => <LoadingOrError hasError={false} onRetry={props.refreshPMtoken} />,
+    sessionToken => {
+      // it should not never happen since this screen is just before the psp selection
+      if (props.pspSelected === null) {
+        return <WorkunitGenericFailure />;
+      }
+      // we have all we need to starts the checkout into the webview
+      const formData = {
+        idPsp: props.pspSelected.id,
+        language: getLocalePrimaryWithFallback(),
+        sessionToken,
+        ...getLookUpIdPO()
+      };
+      return (
+        <PayWebViewModal
+          testID={"PayWebViewModalTestID"}
+          showInfoHeader={false}
+          postUri={urlPrefix + payUrlSuffix}
+          formData={formData}
+          isVisible={!isOnboardingCompleted}
+          finishPathName={webViewExitPathName}
+          onFinish={outcomeCode => {
+            setIsOnboardingComplete(true);
+            props.onCheckoutCompleted(outcomeCode);
+          }}
+          outcomeQueryparamName={webViewOutcomeParamName}
+          onGoBack={props.goBack}
+          modalHeaderTitle={I18n.t("wallet.onboarding.paypal.headerTitle")}
+        />
+      );
+    },
+    _ => <LoadingOrError hasError={true} onRetry={props.refreshPMtoken} />
+  );
+};
+
+/**
+ * This screen includes a webview where the paypal checkout happens. This flow is external to IO, it happens in the Payment Manager
+ * As first step it asks for a fresh token from the Payment Manager, it will be included in the webview
+ * 1. request for a fresh PM token
+ * 2. when the PM token is obtained, starts the checkout challenge in the webview
+ * 3. handle the outcome code coming from the step 2
+ * 4. navigate to the checkout completed screen
+ */
+const PayPalOnboardingCheckoutScreen = (props: Props) => {
+  const navigation = useContext(NavigationContext);
+  const { refreshPMtoken } = props;
+  // refresh the PM at the startup
+  useEffect(() => {
+    refreshPMtoken();
+  }, [refreshPMtoken]);
+
+  const handleCheckoutCompleted = (outcomeCode: Option<string>) => {
+    props.setOutcomeCode(outcomeCode);
+    navigation.dispatch(navigateToPayPalCheckoutCompleted());
+  };
+
+  return (
+    <BaseScreenComponent
+      goBack={props.goBack}
+      headerTitle={I18n.t("wallet.onboarding.paypal.headerTitle")}
+      contextualHelp={emptyContextualHelp}
+    >
+      <CheckoutContent
+        {...props}
+        onCheckoutCompleted={handleCheckoutCompleted}
+      />
+    </BaseScreenComponent>
+  );
+};
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  goBack: () => dispatch(walletAddPaypalBack()),
+  cancel: () => dispatch(walletAddPaypalCancel()),
```

Comment:
```suggestion
```
This seems to be not used

Reply:
☝🏻 
thank you
done here 1f8ffd3657255ae84da12368119e49a22da36616